### ISSUE TITLE
`ref_as_ptr` not available in Rust 1.77 despite documentation

### ISSUE THREAD

**Issue by Rua** [rust-lang/rust-clippy#12559]
### Summary

The documentation for the `ref_as_ptr` lint says that it's available since Rust 1.77, but when I use it, I get a warning to say it's not.

### Reproducer

I tried this code in Cargo.toml:

```rust
[lints]
clippy.ref_as_ptr = "warn"
```

This happened:

```
$ cargo clippy
warning[E0602]: unknown lint: `clippy::ref_as_ptr`
  |
  = help: did you mean: `clippy::ptr_as_ptr`
  = note: requested on the command line with `-W clippy::ref_as_ptr`
  = note: `#[warn(unknown_lints)]` on by default
```

### Version

```text
rustc 1.77.0 (aedd173a2 2024-03-17)
binary: rustc
commit-hash: aedd173a2c086e558c2b66d3743b344f977621a7
commit-date: 2024-03-17
host: x86_64-unknown-linux-gnu
release: 1.77.0
LLVM version: 17.0.6
```

**Comment by J-ZhengLi** [comment]
The version in documentation indicates when the lint **starts development**.

so, in this case, this lint starts its development from nightly 1.77 but finally got merged on Feburary, which did not make it into rust stable yet...

**Comment by Rua** [comment]
Ah I see. A tiny bit confusing to say the least!

**Comment by y21** [comment]
The version for `ref_as_ptr` was corrected to 1.78 in the last changelog (#12735) and is also available in 1.78 so I think we can close this.
